onnxruntime/core/providers/qnn/builder/opbuilder/simple_op_builder.cc

```diff
@@ -261,10 +267,118 @@ Status SimpleOpBuilder::ProcessAttributesAndOutputs(QnnModelWrapper& qnn_model_w
     ORT_RETURN_IF_ERROR(ProcessGridSampleAttributes(qnn_model_wrapper, node_unit, param_tensor_names));
   }
 
-  ORT_RETURN_IF_ERROR(ProcessOutputs(qnn_model_wrapper, node_unit,
-                                     std::move(input_names),
-                                     std::move(param_tensor_names),
-                                     logger, do_op_validation, GetQnnOpType(op_type)));
+  if (op_type == "Sigmoid" || op_type == "Tanh") {
+    // QNN requires 16-bit QDQ Sigmoid and Tanh to use specific output scale and zero-point values
+    // regardless of floating-point range.
+    return ProcessSigmoidOrTanhOutput(qnn_model_wrapper,
```

Comment:
From the QNN docs, Sigmoid and Tanh are the only ops that require specific quantization params. If this changes in the future, we can add a more generic hook that allows op builders to override quant params.